tests/fracs/test_msh_2_grid.py

```diff
@@ -0,0 +1,308 @@
+
+import gmsh
+
+import porepy as pp
+from porepy.fracs.fracture_network_2d import FractureNetwork2d
+from porepy.fracs.fracture_network_3d import FractureNetwork3d
+from porepy.fracs.msh_2_grid import (
+    create_0d_grids,
+    create_1d_grids,
+    create_2d_grids,
+    create_3d_grids,
+)
+from porepy.fracs.simplex import _read_gmsh_file
+from porepy.grids.mdg_generation import _preprocess_simplex_args
+
+dirname: pathlib.Path = pathlib.Path(__file__).parent
+
+
+INCLUSION_NAME: str = "inclusion"
+
+
+@pytest.fixture(scope="function")
+def dims(request: pytest.FixtureRequest) -> tuple:
+    return request.param
+
+
+@pytest.fixture(scope="function")
+def fracture_network(dims: tuple) -> pp.fracture_network:
+    if len(dims) == 2:
+        bounding_box = {
+            "xmin": 0,
+            "xmax": dims[0],
+            "ymin": 0,
+            "ymax": dims[1],
+        }
+        # Two intersecting line fractures.
+        fractures = pp.fracture_sets.orthogonal_fractures_2d(size=dims[0])
+
+    if len(dims) == 3:
+        bounding_box = {
+            "xmin": 0,
+            "xmax": dims[0],
+            "ymin": 0,
+            "ymax": dims[1],
+            "zmin": 0,
+            "zmax": dims[2],
+        }
+        # Three intersecting plane fractures creating both a line fracture intersection
+        # and a point fracture intersection.
+        fractures = pp.fracture_sets.orthogonal_fractures_3d(size=dims[0])
+    domain = pp.Domain(bounding_box)
+    return pp.create_fracture_network(fractures, domain)
+
+
+@pytest.fixture(scope="function")
+def create_gmsh_file(
+    fracture_network: pp.fracture_network, tmp_path: pathlib.Path
+) -> str:
+    """Create a gmsh file of the specified fracture network, with an inclusion
+    added.
+
+    The inclusion is a square in 2d and a cube in 3d, placed somewhere in the middle of
+    the domain. The inclusion is assigned a physical name, ``INCLUSION_NAME``.
+
+    Parameters:
+        fracture_network: The fracture network to create a gmsh file for.
+        tmp_path: Temporary path to store the gmsh file.
+
+    Returns:
+        The path to the created gmsh file.
+
+    """
+    # The idea behind the this function is to create a gmsh geometry for the given
+    # fracture network. Then we will add an inclusion, in the form of a square in 2d and
+    # a cube in 3d, to the geometry, and finally construct the mesh. The resulting gmsh
+    # mesh will have a physical name for the inclusion, which should be picked up
+    # during PorePy mesh generation from the gmsh file.
+
+    msh_file: pathlib.Path = tmp_path / "test.msh"
+
+    # Step 1: Create a gmsh geometry for the fracture network.
+    # Use functionality from pp.create_mdg and fracture_network.mesh to create a gmsh
+    # file from the fracture_network.
+
+    # The following are going to get shifted to extra_args by _preprocess_simplex_args.
+    kwargs: dict = {"file_name": str(msh_file), "write_geo": False}
+
+    dim: int
```

Comment:
Can't this be done in line 109 as dim: int = 2?

Reply:
Yes, who knows why it is there.